core-authentication/src/main/java/gg/solarmc/loader/authentication/LoginAttempt.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * DataLoader
+ * Copyright © 2021 SolarMC Developers
+ *
+ * DataLoader is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * DataLoader is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with DataLoader. If not, see <https://www.gnu.org/licenses/>
+ * and navigate to version 3 of the GNU Affero General Public License.
+ */
+
+package gg.solarmc.loader.authentication;
+
+import java.util.Objects;
+
+public final class LoginAttempt {
+
+	private final ResultType resultType;
+	private final VerifiablePassword verifiablePassword;
+
+	private LoginAttempt(ResultType resultType, VerifiablePassword verifiablePassword) {
+		this.resultType = Objects.requireNonNull(resultType);
+		this.verifiablePassword = verifiablePassword;
+	}
+
+	static LoginAttempt premiumPermitted() {
```

Comment:
Can't you make these static fields so you don't have so many objects just floating around? It doesn't break encapsulation any less than these already do and i see no downside to it

Reply:
We're never going to run out of memory because of objects like LoginAttempt. Short-lived result-transferring objects are the happy optimized path of garbage collectors.

Adding static fields for all of them is just unnecessary logic to review, when you come back to read the code a few months later.